Content.Server/Anomaly/AnomalySystem.cs

```diff
@@ -144,4 +185,74 @@ public override void Update(float frameTime)
         UpdateGenerator();
         UpdateVessels();
     }
+
+    #region Behavior
+    private string GetRandomBehavior()
+    {
+        var weightList = _prototype.Index<WeightedRandomPrototype>(WeightListProto);
+        return weightList.Pick(_random);
+    }
+
+    private void SetBehavior(Entity<AnomalyComponent> anomaly, ProtoId<AnomalyBehaviorPrototype> behaviorProto)
+    {
+        if (anomaly.Comp.CurrentBehavior == behaviorProto)
+            return;
+
+        if (anomaly.Comp.CurrentBehavior != null)
+            RemoveBehavior(anomaly);
+
+        //event broadcast
+        var ev = new AnomalyBehaviorChangedEvent(anomaly, anomaly.Comp.CurrentBehavior, behaviorProto);
+        anomaly.Comp.CurrentBehavior = behaviorProto;
+        RaiseLocalEvent(anomaly, ref ev, true);
+
+        var behavior = _prototype.Index(behaviorProto);
+
+        foreach (var (name, entry) in behavior.Components)
+        {
+            var reg = _componentFactory.GetRegistration(name);
+
+            if (_entity.HasComponent(anomaly, reg.Type))
+            {
+                _entity.RemoveComponent(anomaly, reg.Type);
+            }
+
+            var comp = (Component) _componentFactory.GetComponent(reg);
+            comp.Owner = anomaly;
+
+            var temp = (object) comp;
+            _serialization.CopyTo(entry.Component, ref temp);
+            _entity.RemoveComponent(anomaly, temp!.GetType());
+            _entity.AddComponent(anomaly, (Component) temp!);
+        }
+    }
+
+    private void RemoveBehavior(Entity<AnomalyComponent> anomaly)
+    {
+        if (anomaly.Comp.CurrentBehavior == null)
+            return;
+
+        var behavior = _prototype.Index(anomaly.Comp.CurrentBehavior.Value);
+
+        var entityPrototype = MetaData(anomaly).EntityPrototype;
+        var toRemove = behavior.Components.Keys.ToList();
+
+        foreach (var name in toRemove)
+        {
+            // if the entity prototype contained the component originally
+            if (entityPrototype?.Components.TryGetComponent(name, out var entry) ?? false)
+            {
+                var comp = (Component) _componentFactory.GetComponent(name);
+                comp.Owner = anomaly;
+                var temp = (object) comp;
+                _serialization.CopyTo(entry, ref temp);
+                _entity.RemoveComponent(anomaly, temp!.GetType());
+                _entity.AddComponent(anomaly, (Component) temp);
+                continue;
+            }
+
+            _entity.RemoveComponentDeferred(anomaly, _componentFactory.GetRegistration(name).Type);
+        }
```

Comment:
So resetting an entity completely to its prototype is extremely sussy (stuff like damage, transform, etc) or whatever won't be copied, but if we really really somehow need this then just use removecomponents and https://github.com/space-wizards/RobustToolbox/pull/4934 to copy all the components again.